.github/workflows/deploy.yaml

```diff
@@ -18,18 +18,55 @@ jobs:
     steps:
     - uses: actions/checkout@v2
 
+    - name: Setup Cache
+      uses: actions/cache@v2
+      # NOTE: change key "jupyterbook-N+1" to force rebuilding cache
+      with:
+        path: ./book/_build
+        key: jupyterbook-4
+
+    # NOTE: download build-artifact (_build) folder to inspect cache locally 
+    # - name: List Jupyter-Cache contents
+    #  run: |
+    #    pip install tabulate --no-deps
+    #    jcache cache list -p book/_build/.jupyter_cache
+
     - name: Pull Docker Image
       run: |
         docker pull $DOCKER_IMAGE
         docker images
 
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: us-west-2
+        role-to-assume: github-actions-role
+        role-duration-seconds: 1800
+
     - name: Build JupyterBook
+      env:
+        # Warning: careful exposing secrets in scripts
+        EARTHDATA_LOGIN: ${{ secrets.EARTHDATA_LOGIN }}
+        EARTHDATA_PASSWORD: ${{ secrets.EARTHDATA_PASSWORD }}
```

Comment:
if we put this in, the 'template' becomes less generic... if we're just working with geoscience (and NASA-oriented) events it's helpful, but this is the sort of configuration that is somewhat bespoke. since icesat2 2022 is up next merge it in, and revisit removing things later?

Reply:
That sounds great (leaving it for now and maybe removing later). How NSIDC/NASA wants us to handle cases like this where a login is required for automated rendering/use of open science materials is a discussion I'm currently getting going with some folks at NSIDC, so hopefully we'll get more clarity on that soon!